src/EFCore/Metadata/Internal/ComplexType.cs

```diff
@@ -364,6 +380,29 @@ public override IEnumerable<PropertyBase> FindMembersInHierarchy(string name)
         => FindPropertiesInHierarchy(name)
             .Concat<PropertyBase>(FindComplexPropertiesInHierarchy(name));
 
+    /// <summary>
+    ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
+    ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
+    ///     any release. You should only use it directly in your code with extreme caution and knowing that
+    ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+    /// </summary>
+    public override PropertyCounts Counts
+    {
+        get
+        {
+            Check.DebugAssert(ComplexProperty.IsCollection, $"ComplexType {Name} is not a collection");
+
+            if (_counts == null)
+            {
+                // This will calculate the counts for all contained complex collections
+                var _ = ContainingEntityType.Counts;
```

Comment:
This is a smell :) Consider changing Counts from property to e.g. a CalculateCounts() method?
```suggestion
                _ = ContainingEntityType.Counts;
```